contracts/strategies/examples/assurance-contract/AssuranceContract.sol

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.19;
+
+import {BaseStrategy} from "../../BaseStrategy.sol";
+import {IAllo} from "../../../core/interfaces/IAllo.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @title AssuranceContract
+/// @notice This contract implements an assurance contract strategy for crowdfunding, supporting both ETH and ERC20 tokens.
+/// @dev Extends BaseStrategy to integrate with the Allo protocol.
+contract AssuranceContract is BaseStrategy {
+    using SafeERC20 for IERC20;
+
+    /// @notice Struct to store campaign details
+    /// @param goal The funding goal of the campaign
+    /// @param totalPledged Total amount pledged so far
+    /// @param deadline Timestamp when the campaign ends
+    /// @param beneficiary Address to receive funds if goal is met
+    /// @param finalized Whether the campaign has been finalized
+    /// @param tokenAddress Address of ERC20 token, or address(0) for ETH
+    struct Campaign {
+        uint256 goal;
+        uint256 totalPledged;
+        uint256 deadline;
+        address beneficiary;
+        bool finalized;
+        address tokenAddress;
+    }
+
+    /// @notice Mapping of pool IDs to Campaign structs
+    mapping(uint256 => Campaign) public campaigns;
```

Comment:
I see you have this mapping here, does this mean the strategy would be re-usable for multiple campaigns after each one finishes? If so it would be cleaner to also include a `createCampaign` function for pool managers that will basically create the campaign like you do in `initialize`

Reply:
i believe `pool ID` here is also reffering to the ID of the campaign, this might confuse people with Allo's pool IDs